.github/workflows/main.yml

```diff
@@ -0,0 +1,134 @@
+name: CI Pipeline
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'python-flask/**'
+      - 'c-sharp-razor/**'
+      - 'rust-actix/**'
+      - 'ruby-sinatra/**'
+      - 'javascript-express/**'
+      - 'go-gin/**'
+      - 'go-gorilla/**'
+      - '/**'
+
+jobs:
+  check-root-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      root_changed: ${{ steps.set-flags.outputs.root_changed }}
+      service_changed: ${{ steps.set-flags.outputs.service_changed }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check if root files or any service changed
+        id: set-flags
+        run: |
+          # Initialize flags
+          root_changed="false"
+          any_service_changed="false"
+
+          # Get the list of changed files by comparing against the base branch
+          git fetch origin main
+          changed_files=$(git diff --name-only origin/main...HEAD)
+
+          # Check if any of the changed files are outside the known service directories
+          for file in $changed_files; do
+            if [[ ! "$file" =~ ^(python-flask/|c-sharp-razor/|rust-actix/|ruby-sinatra/|javascript-express/|go-gin/|go-gorilla/|\.github/) ]]; then
+              root_changed="true"
+              echo "Root-level file changed: $file"
```

Comment:
I think you can remove this echo

Reply:
It is nice to see what you expected triggered the tests no? :)

Reply:
I think not at this level. The PR has this information already. It is ok to keep it while debugging the CI during the implementation.

Reply:
deleted